src/lib/SaferNames/Type.hs

```diff
@@ -1013,6 +975,21 @@ applyDataDefParams :: ScopeReader m => DataDef n -> [Type n] -> m n [DataConDef
 applyDataDefParams (DataDef _ bs cons) params =
   fromListE <$> applyNaryAbs (Abs bs (ListE cons)) (map SubstVal params)
 
+checkedApplyDataDefParams :: (BindingsReader m, Fallible1 m) => DataDef n -> [Type n] -> m n [DataConDef n]
+checkedApplyDataDefParams (DataDef _ bs cons) params =
+  fromListE <$> checkedApplyNaryAbs (Abs bs (ListE cons)) params
+
+-- TODO: Subst all at once, not one at a time!
```

Comment:
The type checking monad carries a substitution so we can use that.  We would then just call `runEnvReaderT`  with the empty substitution from `checkedApplyDataDefParams`